From 035a4db85bbf28ba1452c49c9f8d05a085f2544b Mon Sep 17 00:00:00 2001
From: Prabhakar Kushwaha <prabhakar.kushwaha@nxp.com>
Date: Tue, 26 Apr 2016 17:40:05 +0530
Subject: [PATCH 18/93] boards: ls1012aqds: Enable SDHC_CD in brdcfg10 of FPGA

Default configuration of brdcfg10 in FPGA does not enable SDHC_CD
signal.

Enable SDHC_CD by default during boot sequence.

Signed-off-by: Calvin Johnson <calvin.johnson@nxp.com>
Signed-off-by: Prabhakar Kushwaha <prabhakar.kushwaha@nxp.com>
---
 board/freescale/ls1012aqds/ls1012aqds.c |   12 ++++++++++++
 include/configs/ls1012aqds.h            |    3 +++
 2 files changed, 15 insertions(+)

diff --git a/board/freescale/ls1012aqds/ls1012aqds.c b/board/freescale/ls1012aqds/ls1012aqds.c
index ffcd0d8..6d5fef8 100644
--- a/board/freescale/ls1012aqds/ls1012aqds.c
+++ b/board/freescale/ls1012aqds/ls1012aqds.c
@@ -154,6 +154,18 @@ int board_early_init_f(void)
 	return 0;
 }
 
+#ifdef CONFIG_MISC_INIT_R
+int misc_init_r(void)
+{
+	u8 mux_sdhc_cd = 0x80;
+
+	i2c_set_bus_num(0);
+
+	i2c_write(CONFIG_SYS_I2C_FPGA_ADDR, 0x5a, 1, &mux_sdhc_cd, 1);
+	return 0;
+}
+#endif
+
 int board_init(void)
 {
 	struct ccsr_cci400 *cci = (struct ccsr_cci400 *)
diff --git a/include/configs/ls1012aqds.h b/include/configs/ls1012aqds.h
index fdada18..bb433de 100644
--- a/include/configs/ls1012aqds.h
+++ b/include/configs/ls1012aqds.h
@@ -27,6 +27,7 @@
 #endif
 
 #define CONFIG_QIXIS_I2C_ACCESS
+#define CONFIG_SYS_I2C_FPGA_ADDR	0x66
 
 /*
  * I2C bus multiplexer
@@ -145,4 +146,6 @@
 #define CONFIG_SYS_MEMTEST_START	0x80000000
 #define CONFIG_SYS_MEMTEST_END		0x9fffffff
 
+#define CONFIG_MISC_INIT_R
+
 #endif /* __LS1012AQDS_H__ */
-- 
1.7.9.5

